### PR TITLE
Basic fire store communication

### PR DESCRIPTION
**Pull request by BartLucien**
This PR add a database class which exposes basic function to communicate with the fireStore Server.

> Add second PR because the first one is bugging.

### CODE REVIEW

File: app/src/main/java/com/github/fribourgsdp/radio/FireBaseTestActivity.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class FireBaseTestActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_fire_base_test)
+
+
+        val db = Database()
+
+        val userAuthUID = "m0sd9l"//FirebaseAuth.getInstance().currentUser?.uid
```
Comment (VictorSchneuwly) [inline app/src/main/java/com/github/fribourgsdp/radio/FireBaseTestActivity.kt:20]: Maybe you should remove the old commented code.

Comment (BartLucien) [reply app/src/main/java/com/github/fribourgsdp/radio/FireBaseTestActivity.kt]: I added a clarification. This code will be used later.

File: app/build.gradle [hunk 2]
```diff
@@ -37,25 +38,53 @@ android {
...
+
 dependencies {
 
     implementation 'androidx.core:core-ktx:1.7.0'
     implementation 'androidx.appcompat:appcompat:1.4.1'
     implementation 'com.google.android.material:material:1.5.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.3'
+
+
+    implementation platform('com.google.firebase:firebase-bom:29.1.0')
```
[the remark this answers is not in the document]
Comment (BartLucien) [reply app/build.gradle:65]: I put the newer version

File: app/src/androidTest/java/com/github/fribourgsdp/radio/FireBaseTestActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.junit.Rule
+import org.junit.Test
+
+class FireBaseTestActivityTest {
+
+    @get:Rule
+    var FireBaseTestRule = ActivityScenarioRule(FireBaseTestActivity::class.java)
+
+    //this class is to test the Database feature easily, it doesn't have any functionality so I just run the activity to have the coverage
+    @Test
```
[the remark this answers is not in the document]
Comment (BartLucien) [reply app/src/androidTest/java/com/github/fribourgsdp/radio/FireBaseTestActivityTest.kt:14]: No it is just a way to get the coverage for the activity that I created to test the Database functions

**Review by sim0mo — APPROVED** [review]

**Review by Flamantrose321 — APPROVED** [review]
Everything seems ok

**Pull request merged**